### ISSUE TITLE
Flake Newbie: How to add a local flake to packages/devshells

### ISSUE THREAD

**Issue by bennyandresen** [Misterio77/nix-starter-configs#25]
hey

very cool starter config to get familiar with flakes for NixOS. (also home-manager which I haven't used before.)
Before I start: my legacy packages work fine using the `pkgs/` structure. Directories with no `default.nix` and only a `flake.nix` can't be configured that way however.

Question: How do I add flakes that export a defaultPackage to this configuration?

```
inputs = {
...
    # custom
    bbterm = {
      url = "path:/home/benny/Code/bbterm";
      inputs.nixpkgs.follows = "nixpkgs";
    };
...
};
```

`nix run /home/benny/Code/bbterm` works fine, but downloads the nixpkgs tarball every time. I would like to have it integrated into the built `nixosSystem`

Thanks in advance!

**Comment by Misterio77** [comment]
Hi! Thanks for your kind words :D

Okay so, there's basically two ways to add a package to your configuration. You can either:

- Add an overlay to your nixpkgs instance (then you would access it same as a regular `pkgs.foo-bar`)
```nix
{ inputs, pkgs, ... }: {
  nixpkgs.overlays = [
    (final: prev: {
      bbterm = inputs.bbterm.defaultPackage.${pkgs.system};
    })
  ];

  environment.systemPackages = [
    pkgs.bbterm
  ];
}
```
- Use `inputs.bbterm.packages.${pkgs.system}.name` anywhere in your NixOS Configuration that takes a package (for example, `environment.systemPackages`).
```nix
{ inputs, pkgs, ... }: {
  environment.systemPackages = [
    inputs.bbterm.defaultPackage.${pkgs.system}
  ];
}
```

**Comment by bennyandresen** [comment]
Over the weekend I've found option 2 by playing around a bit but to have it be cleaner I went with option 1.

I added
```
(final: prev: {
      bbterm = inputs.bbterm.defaultPackage.${pkgs.system};
    })
```
to the overlays in [./home-manager/home.nix](https://github.com/Misterio77/nix-starter-configs/blob/main/standard/home-manager/home.nix) and could then reference it in the same file in `home.packages`.

I assume that's where most people that use home-manager put their programs, so that's where I will put my user specific programs from now on. :-)
